ocaml/xapi/repository.ml

```diff
@@ -1024,3 +1024,128 @@ let get_pool_updates_in_json ~__context ~hosts =
   | e ->
     error "getting updates for pool failed: %s" (ExnHelper.string_of_exn e);
     raise Api_errors.(Server_error (get_updates_failed, []))
+
+let apply ~__context ~host =
+  (* This function runs on slave host *)
+  with_local_repository ~__context (fun () ->
+      let params =
+          [
+            "-y"; "--disablerepo=*";
+            Printf.sprintf "--enablerepo=%s" !Xapi_globs.local_repo_name;
+            "upgrade"
+          ]
+      in
+      try ignore (Helpers.call_script !Xapi_globs.yum_cmd params)
+      with e ->
+        let ref = Ref.string_of host in
+        error "Failed to apply updates on host ref='%s': %s" ref (ExnHelper.string_of_exn e);
+        raise Api_errors.(Server_error (apply_updates_failed, [ref])))
+
+let set_of_list l =
+  l
+  |> List.map string_of_guidance
+  |> GuidanceStrSet.of_list
+
+let set1 = set_of_list [EvacuateHost; RestartToolstack]
+
+let set2 = set_of_list [RestartDeviceModel; RestartToolstack]
+
+let set3 = set_of_list [RestartDeviceModel; EvacuateHost]
+
+let set4 = set_of_list [EvacuateHost; RestartToolstack; EvacuateHost]
+
+let assert_valid_guidances gs =
+  match List.map (fun x -> guidance_of_string x) gs with
+  | [RebootHost]
+  | [EvacuateHost]
+  | [RestartToolstack]
+  | [RestartDeviceModel] -> ()
+  | l when GuidanceStrSet.equal (set_of_list l) set1 -> ()
```

Comment:
There are several calls `GuidanceStrSet.equal (set_of_list l) set1 ` - so this could be abstracted and used subsequently:

```
let eq list set = GuidanceStrSet.equal (set_of_list list) set